packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -200,43 +173,49 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
 
   Future<PutLogEventsResponse> _sendToCloudWatch(
     List<InputLogEvent> logEvents,
-    String logStreamName,
   ) async {
+    final logStreamName = await _logStreamProvider.logStream;
     final request = PutLogEventsRequest(
       logGroupName: _pluginConfig.logGroupName,
       logStreamName: logStreamName,
       logEvents: logEvents,
     );
-    return _client.putLogEvents(request).result;
+    try {
+      return _client.putLogEvents(request).result;
+    } on ResourceNotFoundException {
+      await _logStreamProvider.createLogStream(logStreamName);
+      return _client.putLogEvents(request).result;
```

Comment:
```suggestion
      return await _client.putLogEvents(request).result;
```

Reply:
it is unnecessary await in return: https://dart.dev/tools/linter-rules/unnecessary_await_in_return

Reply:
It's a bug: https://github.com/dart-lang/linter/issues/2357

Reply:
Move the return to the next line to avoid lint

Reply:
But the return is actually not needed in this case

Reply:
The await is

Reply:
Hmm yes, sorry, I thought we were speaking about the `try` block. My mistake, there is no semantic difference for this line.